### ISSUE TITLE
Project failed with mongodb atlas when using mongo+srv://

### ISSUE THREAD

**Issue by Duckybee** [ForestAdmin/lumber#222]
There is issue with connecting to mongodb atlas where they using mongo+srv for connection. After debugging into this last night, it appears that we didnt take mongo+srv into consideration in few places. I have the fix locally and it works for me but dont have permission to create PR so just post the changes here in case it help.

To repro this:
you will need to create a MongoDb Atlas paid account, go through `lumber generate` and then try `lumber install`
`Lumber generate` does generate the project but it will create the wrong DB Connection string. That will cause issue when we do `lumber install`

To fix the issue:
2. Update lumber-cli/services/db.js (Find Database function)
```
function Database() {
  function error(err) {
    logger.error(
      'Cannot connect to the database due to the following error:',
      err,
    );

    process.exit(1);
  }

  function sequelizeAuthenticate(db) {
    return db.authenticate()
      .then(() => db)
      .catch(err => error(err));
  }

  this.connect = (options) => {
    const isSSL = options.dbSSL || options.ssl;
    let db;

    if (options.dbConnectionUrl) {
      if (options.dbConnectionUrl.startsWith('postgres://')) {
        options.dbDialect = 'postgres';
      } else if (options.dbConnectionUrl.startsWith('mysql://')) {
        options.dbDialect = 'mysql';
      } else if (options.dbConnectionUrl.startsWith('mssql://')) {
        options.dbDialect = 'mssql';
      } else if (options.dbConnectionUrl.startsWith('mongodb://') || options.dbConnectionUrl.startsWith('mongodb+srv://')) {
        options.dbDialect = 'mongodb';
      }
    }

    if (options.dbDialect === 'mongodb') {
      const opts = { useNewUrlParser: true };
      let connectionUrl = options.dbConnectionUrl;

      // This is for lumber-generate to connect to MongoDb
      if (!connectionUrl) {
        connectionUrl = 'mongodb';
        if (options.mongodbSrv) { connectionUrl += '+srv'; }
        connectionUrl += '://';
        if (options.dbUser) { connectionUrl += options.dbUser; }
        if (options.dbPassword) { connectionUrl += `:${options.dbPassword}`; }
        connectionUrl += `@${options.dbHostname}`;
        if (!options.mongodbSrv) { connectionUrl += `:${options.dbPort}`; }
        connectionUrl += `/${options.dbName}`;
        if (isSSL) { opts.ssl = true; }
      }

      return MongoClient.connect(connectionUrl, opts)
        .then(client => client.db(options.dbName));
    }

    const needsEncryption = isSSL && (options.dbDialect === 'mssql');

    const connectionOpts = {
      logging: false,
      dialectOptions: {
        ssl: isSSL,
        encrypt: needsEncryption,
      },
    };

    if (options.dbConnectionUrl) {
      db = new Sequelize(options.dbConnectionUrl, connectionOpts);
    } else {
      connectionOpts.host = options.dbHostname;
      connectionOpts.port = options.dbPort;
      connectionOpts.dialect = options.dbDialect;

      db = new Sequelize(
        options.dbName, options.dbUser,
        options.dbPassword, connectionOpts,
      );
    }

    return sequelizeAuthenticate(db);
  };
}
```
2. Update dumper.js
```
function getDatabaseUrl() {
    let connectionString;

    if (config.dbConnectionUrl) {
      connectionString = config.dbConnectionUrl;
    } else if (config.dbDialect === 'sqlite') {
      connectionString = `sqlite://${config.dbStorage}`;
    } else {
      if (config.dbDialect === 'mongodb' && config.mongodbSrv) {
        connectionString = `${config.dbDialect}+srv://${config.dbUser}`;
        if (config.dbPassword) {
          // NOTICE: Encode password string in case of special chars.
          connectionString += `:${encodeURIComponent(config.dbPassword)}`;
        }
        connectionString += `@${config.dbHostname}/${config.dbName}`;
      } else {
        connectionString = `${config.dbDialect}://${config.dbUser}`;
        if (config.dbPassword) {
          // NOTICE: Encode password string in case of special chars.
          connectionString += `:${encodeURIComponent(config.dbPassword)}`;
        }
        connectionString += `@${config.dbHostname}:${config.dbPort}/${config.dbName}`;
      }
    }
    return connectionString;
  }
```

**Comment by SpekkoRice** [comment]
Are you going make a PR?

**Comment by Duckybee** [comment]
I dont have permission to create the PR so I just pushed the change there in case someone wants to help me bring it it.

**Comment by SpekkoRice** [comment]
Well I forked and patched it here to be installed on our servers: https://github.com/SpekkoRice/lumber if anyone is interested.

**Comment by arnaudbesnier** [comment]
Hi @Duckybee @SpekkoRice thanks for the feedback.

This repository is open-source so you can submit pull requests once the repo forked and the fix applied.
I'll try to integrate that fix today based on your fork @SpekkoRice 🙏

**Comment by arnaudbesnier** [comment]
Fixed here: https://github.com/ForestAdmin/lumber/commit/d96ccc2a286530f7c23ed5bc91ae92efc9744255

The fix should be released in a Lumber v2.2.0.
